extensions/amp-story/0.1/test/test-amp-story-info-dialog.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Action,
+  AmpStoryStoreService,
+  StateProperty,
+} from '../amp-story-store-service';
+import {
+  DIALOG_VISIBLE_CLASS,
+  InfoDialog,
+  MOREINFO_VISIBLE_CLASS,
+} from '../amp-story-info-dialog';
+import {Services} from '../../../../src/services';
+import {registerServiceBuilder} from '../../../../src/service';
+
+
+describes.realWin('amp-story-share-menu', {amp: true}, env => {
+  let messagingReadyPromise;
+  let moreInfoLinkUrl;
+  let parentEl;
+  let infoDialog;
+  let storeService;
+  let win;
+
+  const MOREINFO_CLASS = '.i-amphtml-story-info-moreinfo';
+
+  beforeEach(() => {
+    win = env.win;
+    storeService = new AmpStoryStoreService(win);
+    registerServiceBuilder(win, 'story-store', () => storeService);
+
+    // Making sure the vsync tasks run synchronously.
+    sandbox.stub(Services, 'vsyncFor').returns({
+      mutate: fn => fn(),
+      runPromise: (vsyncTaskSpec, vsyncState) => {
+        vsyncTaskSpec.measure(vsyncState);
+        vsyncTaskSpec.mutate(vsyncState);
+        return Promise.resolve();
+      },
+    });
+
+    sandbox.stub(Services, 'localizationServiceV01').returns({
+      getLocalizedString: localizedStringId => `string(${localizedStringId})`,
+    });
+
+    sandbox.stub(Services, 'viewerForDoc').returns({
+      whenMessagingReady: () => messagingReadyPromise,
+      sendMessageAwaitResponse: eventType => {
+        if (eventType === 'moreInfoLinkUrl') {
+          return Promise.resolve(moreInfoLinkUrl);
+        }
+
+        return Promise.resolve();
+      },
+    });
+
+    parentEl = win.document.createElement('div');
+    win.document.body.appendChild(parentEl);
+    infoDialog = new InfoDialog(win, parentEl);
+  });
+
+  it('should build the info dialog', () => {
+    infoDialog.build().then(() => {
```

Comment:
How does mocha know it has to wait to run the assertions here and below? Do we have to return the promise?

Reply:
Whoops :smile: